### PR TITLE
Added materials to CoalGenerator

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description
Tags
please dont break it this time <3

## Changes
added materials to coalgen

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [x] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java [hunk 1]
```diff
@@ -22,6 +24,11 @@ protected void registerDefaultFuelTypes() {
         registerFuel(new MachineFuel(12, new ItemStack(Material.BLAZE_ROD)));
         registerFuel(new MachineFuel(20, new ItemStack(Material.DRIED_KELP_BLOCK)));
 
+        // Boats
+        for (Material mat : Tag.ITEMS_BOATS.getValues()) {
+            registerFuel(new MachineFuel(6, new ItemStack(mat)));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:29]: You use 5 planks to craft a boat, so it should be 5 here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java [hunk 2]
```diff
@@ -35,8 +42,44 @@ protected void registerDefaultFuelTypes() {
...
+        }
+
+        // Wooden Fences
+        for (Material mat : Tag.WOODEN_FENCES.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        // wooden Trapdoors
+        for (Material mat : Tag.WOODEN_TRAPDOORS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:63]: You use 6 planks to craft 2 trapdoors, so it should be 3 here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java [hunk 3]
```diff
@@ -35,8 +42,44 @@ protected void registerDefaultFuelTypes() {
...
+        }
+
+        // Wooden Pressure Plates
+        for (Material mat : Tag.WOODEN_PRESSURE_PLATES.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:68]: You use 2 planks to craft a pressure plate, so it should be 2 here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java [hunk 4]
```diff
@@ -35,8 +42,44 @@ protected void registerDefaultFuelTypes() {
...
+        }
+
+        // Wooden Doors
+        for (Material mat : Tag.WOODEN_DOORS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:73]: You use 6 planks to craft 3 doors, so it should be 2 here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java [hunk 5]
```diff
@@ -35,8 +42,44 @@ protected void registerDefaultFuelTypes() {
...
+        }
+
+        // Signs
+        for (Material mat : Tag.SIGNS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:78]: You use 6 planks to craft 3 signs, so it should be 2 here.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 1 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:29]: i got this info straight from the wiki https://minecraft.gamepedia.com/Smelting im fine changing it if you actually want to have smelting time differ from mc official time

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 2 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:63]: i got this info straight from the wiki https://minecraft.gamepedia.com/Smelting im fine changing it if you actually want to have smelting time differ from mc official time

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 4 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:73]: i got this info straight from the wiki https://minecraft.gamepedia.com/Smelting im fine changing it if you actually want to have smelting time differ from mc official time

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 5 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:78]: i got this info straight from the wiki https://minecraft.gamepedia.com/Smelting im fine changing it if you actually want to have smelting time differ from mc official time

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 3 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:68]: i got this info straight from the wiki https://minecraft.gamepedia.com/Smelting im fine changing it if you actually want to have smelting time differ from mc official time

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java [hunk 6]
```diff
@@ -73,18 +72,58 @@ public static PlayerProfile awaitProfile(OfflinePlayer player) throws Interrupte
...
+            Material.DARK_OAK_BOAT, Material.JUNGLE_BOAT, Material.OAK_BOAT, Material.SPRUCE_BOAT
+        );
+        server.createMaterialTag(NamespacedKey.minecraft("wooden_buttons"), Material.BIRCH_BUTTON,
+            Material.ACACIA_BUTTON, Material.SPRUCE_BUTTON, Material.DARK_OAK_BUTTON,
+            Material.JUNGLE_BOAT, Material.OAK_BUTTON
+        );
+        server.createMaterialTag(NamespacedKey.minecraft("signs"), Material.SPRUCE_SIGN, Material.ACACIA_SIGN,
+            Material.OAK_SIGN, Material.JUNGLE_SIGN, Material.DARK_OAK_SIGN, Material.BIRCH_SIGN
+        );
+        server.createMaterialTag(NamespacedKey.minecraft("doors"), Material.SPRUCE_DOOR, Material.ACACIA_DOOR,
```
Comment (TheBusyBiscuit) [inline src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:120]: This should be "wooden_doors".

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:29]: I'd rather have them make sense than be accurate :LUL:

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 2 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:63]: I'd rather have them make sense than be accurate :LUL:

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 3 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:68]: I'd rather have them make sense than be accurate :LUL:

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 4 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:73]: I'd rather have them make sense than be accurate :LUL:

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java (hunk 5 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/generators/CoalGenerator.java:78]: I'd rather have them make sense than be accurate :LUL:

**Comment by J3fftw1** [comment]
shouldnt logs be 4 then instead of 1 since you get 4 planks per log?

**Comment by TheBusyBiscuit** [comment]
Probably yeah

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java [hunk 7]
```diff
@@ -73,18 +72,58 @@ public static PlayerProfile awaitProfile(OfflinePlayer player) throws Interrupte
...
-        server.createMaterialTag(NamespacedKey.minecraft("leaves"), Material.OAK_LEAVES, Material.BIRCH_LEAVES, Material.SPRUCE_LEAVES, Material.JUNGLE_LEAVES, Material.ACACIA_LEAVES, Material.DARK_OAK_LEAVES);
-        server.createMaterialTag(NamespacedKey.minecraft("saplings"), Material.OAK_SAPLING, Material.BIRCH_SAPLING, Material.SPRUCE_SAPLING, Material.JUNGLE_SAPLING, Material.ACACIA_SAPLING, Material.DARK_OAK_SAPLING);
-        server.createMaterialTag(NamespacedKey.minecraft("coral_blocks"), Material.BRAIN_CORAL_BLOCK, Material.BUBBLE_CORAL_BLOCK, Material.FIRE_CORAL_BLOCK, Material.HORN_CORAL_BLOCK, Material.TUBE_CORAL_BLOCK);
-        server.createMaterialTag(NamespacedKey.minecraft("corals"), Material.BRAIN_CORAL, Material.BUBBLE_CORAL, Material.FIRE_CORAL, Material.HORN_CORAL, Material.TUBE_CORAL);
-        server.createMaterialTag(NamespacedKey.minecraft("ice"), Material.ICE, Material.PACKED_ICE, Material.FROSTED_ICE, Material.BLUE_ICE);
-        server.createMaterialTag(NamespacedKey.minecraft("fire"), Material.FIRE, Material.SOUL_FIRE);
```
Comment (TheBusyBiscuit) [inline src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:87]: You accidentally removed the fire tag, that's why it is failing.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Found the issue

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 7 above)
Comment (J3fftw1) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:87]: Thonks

**Comment by J3fftw1** [comment]
wooohoo all checks passed

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**